code/modules/mob/living/simple_animal/friendly/iriska.dm

```diff
@@ -0,0 +1,180 @@
+/mob/living/simple_animal/iriska
+	name = "Iriska"
+	desc = "The captain's own cat. Fat and lazy."
+	icon = 'icons/mob/iriska.dmi'
+	icon_state = "iriska"
+	health = 80
+	maxHealth = 80
+	wander = FALSE
+	canmove = FALSE
+	speak_emote = list("purrs.", "meows.")
+	emote_see = list("shakes her head.", "shivers.")
+	speak_chance = 0.75
+	meat_amount = 6
+	meat_type = /obj/item/reagent_containers/food/snacks/meat
+	response_help = "pets"
+	response_disarm = "rubs"
+	response_harm = "makes terrible mistake by kicking"
+	min_oxy = 16
+	minbodytemp = 223
+	maxbodytemp = 323
+	mob_size = MOB_HUGE
+	harm_intent_damage = 20
+	melee_damage_lower = 10
+	melee_damage_upper = 30
+	attacktext = "slashed"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	density = TRUE
+	bite_factor = 0.8
+	stomach_size_mult = 10
+
+	autoseek_food = FALSE // Original code for seeking food doesn't fit, Iriska will use slightly modified version
+	beg_for_food = FALSE
+	max_scan_interval = 10
+	eat_from_hand = FALSE
+
+var/atom/snack = null
+
+var/list/tolerated = list()
+var/list/despised = list()
+
+/mob/living/simple_animal/iriska/fall_asleep()
+	return
+
+/mob/living/simple_animal/iriska/wake_up()
+	return
+
+/mob/living/simple_animal/iriska/Life()
+	.=..()
+
+	if(!stasis)
+
+		seek_food()
+		react_to_mob()
+
+/mob/living/simple_animal/iriska/proc/seek_food()
+	turns_since_scan++
+	if(turns_since_scan >= scan_interval)
+		turns_since_scan = 0
+		if(snack && (!(isturf(snack.loc) || (foodtarget && !can_eat()) )))
+			snack = null
+			foodtarget = 0
+		if(!snack || !(snack.loc in oview(src, 1)))
+			snack = null
+			foodtarget = 0
+			if (can_eat())
+				for(var/obj/item/reagent_containers/food/snacks/S in oview(src,1))
+					if(!istype(S, /obj/item/reagent_containers/food/snacks/grown))
+						if(isturf(S.loc))
+							snack = S
+							foodtarget = 1
+							break
+
+		if(snack)
+			scan_interval = min_scan_interval
+
+			if (snack.loc.x < src.x)
+				set_dir(WEST)
+			else if (snack.loc.x > src.x)
+				set_dir(EAST)
+			else if (snack.loc.y < src.y)
+				set_dir(SOUTH)
+			else if (snack.loc.y > src.y)
+				set_dir(NORTH)
+			else
+				set_dir(SOUTH)
+
+			if(isturf(snack.loc) && Adjacent(get_turf(snack), src))
+				var/mob/mob = get_mob_by_key(snack.fingerprintslast)
+				UnarmedAttack(snack)
+				if(!(mob in despised))
+					tolerate(mob)
+	else
+		scan_interval = max(min_scan_interval, min(scan_interval+1, max_scan_interval))
+
+/mob/living/simple_animal/iriska/proc/react_to_mob()
+	for(var/mob/living/M in oview(src, 1))
```

Comment:
``HasProximity()`` would be better for such job

Reply:
Thanks, I'll replace that in next Iriska update, if all works fine.